src/lib/action/entry-derive.ts

```diff
@@ -27,40 +27,68 @@ class EntryDeriveAction extends APIAction {
   async applyTo (api: OfflineAPI) {
     const entries: Entry[] = await api.getEntriesForContentType(this.contentTypeId)
     const locales: string[] = await api.getLocalesForSpace()
+
     for (const entry of entries) {
       const inputs = _.pick(entry.fields, this.fromFields)
       const newEntryId = await this.identityKey(inputs)
       const hasEntry = await api.hasEntry(newEntryId)
 
+      let skipEntry = true
+      let fieldsForTargetEntry = {}
+
+      for (const locale of locales) {
+        let outputsForCurrentLocale
+        try {
+          outputsForCurrentLocale = await this.deriveEntryForLocale(inputs, locale)
+        } catch (err) {
+          await api.recordRuntimeError(err)
+          continue
+        }
+
+        if (typeof outputsForCurrentLocale === 'undefined') {
+          continue
+        }
+
+        skipEntry = false
+
+        // we collect all the values for the target entry before writing it to the
+        // offline API because we don't yet know if the entry might be skipped
+        // TODO: verify that the derivedFields actually get written to
+        // and to no other field
+        Object.keys(outputsForCurrentLocale).forEach((fieldId) => {
```

Comment:
Minor styling:
```js
for ([fieldId, localizedValue] of entries(outputsForCurrentLocale))
```
makes it a bit clearer to me on why we need all `Object.keys` in each iteration.

Reply:
That reads much better indeed